backend/corpora/api_server/app.py

```diff
@@ -164,5 +167,11 @@ def handle_corpora_error(exception):
     )
 
 
+@app.errorhandler(InternalServerError)
+def handle_internal_server_error(exception):
+    app.logger.exception("InternalServerError", exc_info=exception.original_exception)
+    return FlaskApi.get_response(problem(500, "Internal Server Error"))
```

Comment:
From my reading of the docs, this will replace the original response object with a new response object... right? Are we worried about losing the message detail from the error thrown by the code (which this middleware is catching)?

Reply:
Correct, we hide the details of the server error from the user and log it on your side.